### ISSUE TITLE
Icon grey even though you can complete the quest

### ISSUE THREAD

**Issue by Muehe** [fusionpit/QuestIconDesaturation#2]
Hello there.

I noticed the disclaimer you put on your project about finished quests not showing up as completed, and it is probably related to quests that do not have any objectives text. [These quests can be turned in immediately](https://github.com/cmangos/issues/wiki/Quest_template#objectives).

There are ~600 quests for which this is the case, but most of them are "turn in only" quests, many of them repeatable, like [Give Gerard a Drink](https://classic.wowhead.com/quest=16/give-gerard-a-drink), so they never appear in the quest log.

Some of them can be picked up though, just don't have any kind of complete-able objective, and are thus never flagged as `isComplete` by `GetQuestLogTitle`, like [The Hunt Completed](https://classic.wowhead.com/quest=247/the-hunt-completed)

If my assumption is correct, you need to check

`local _, questObjectives = GetQuestLogQuestText();`

for being `nil` (or empty string, not sure which it returns) in [line 32](https://github.com/fusionpit/QuestIconDesaturation/blob/master/QuestIconDesaturation.lua#L32) to fix this bug.

Alternatively you could try checking if `GetNumQuestLeaderBoards` returns 0, or just check both of these variables.

Hope this helps!

**Comment by fusionpit** [comment]
Hi Muehe, thanks for the info!  I did come to the same conclusion as you re: quests lacking objectives, but I found that `IsQuestComplete` actually covers everything I have encountered so far.  `IsQuestComplete` seems to be the same internal logic that is used when showing a grey or gold ? icon over the quest giver's head in the 3d world.  Seems like it was introduced into the game in WoD, but I'm not totally sure.
